faq_automation/rag_agent.py

```diff
@@ -0,0 +1,199 @@
+
+    order: int = Field(..., description="Integer controlling sort order within the module. Set to number if it should be placed near existing FAQ records, set to -1 it it should go at the end of the section")
+
+    question: str = Field(..., description="FAQ question title displayed to users (plain-text question).")
+
+    # Action-specific payload
+    proposed_content: Optional[str] = Field(
+        None,
+        description="Only for NEW and UPDATE: markdown file with the answer. The question is not included. No headers.",
+    )
+
+    filename_slug: Optional[str] = Field(
+        None,
+        description="Only for NEW: file-system friendly slug with hyphens, up to 50 characters",
+    )
+
+    # Notes
+    warnings: List[str] = Field(
+        default_factory=list,
+        description="Optional warnings (e.g., sort order collision, module mismatch).",
+    )
+
+
+class FAQAgent:
+    """Agent for processing FAQ proposals using RAG and LLM"""
+
+    def __init__(self, course_dir: Path, openai_api_key: str, model: str = "gpt-5-nano"):
+        """
+        Initialize the FAQ Agent
+
+        Args:
+            course_dir: Path to the course directory (e.g., _questions/machine-learning-zoomcamp)
+            openai_api_key: OpenAI API key
+            model: OpenAI model to use (default: gpt-4)
+        """
+        self.course_dir = course_dir
+        self.openai_client = OpenAI(api_key=openai_api_key)
+        self.model = model
+
+        # Load existing FAQs and metadata
+        self.documents = read_questions(course_dir)
+        self.metadata = read_metadata(course_dir)
+
+        # Build search index
+        self.index = Index(
+            text_fields=['section', 'question', 'answer'],
+            keyword_fields=['course', 'section_id'],
+        )
+        self.index.fit(self.documents)
+
+    def process_proposal(self, question: str, answer: str, num_results: int = 5) -> FAQDecision:
+        """
+        Process a new FAQ proposal
+
+        Args:
+            question: The proposed question
+            answer: The proposed answer
+            num_results: Number of similar FAQs to retrieve (default: 5)
+
+        Returns:
+            FAQDecision object with action and all necessary information
+        """
+        # Search for similar existing FAQs
+        proposal = f"## {question}\n\n{answer}"
+        results = self.index.search(proposal, num_results=num_results)
+        results = keep_relevant(results)
+
+        # Build prompt
+        prompt = PROMPT_TEMPLATE.format(
+            entry=proposal,
+            results=json.dumps(results),
+            sections=json.dumps(self.metadata['sections'])
+        )
+
+        messages = [
+            {"role": "system", "content": SYSTEM_PROMPT},
+            {"role": "user", "content": prompt}
+        ]
+
+        # Call OpenAI with structured output
+        response = self.openai_client.beta.chat.completions.parse(
```

Comment:
Should be the responses API like in the notebook 

How much of the Python code from the notebook changed?